### ISSUE TITLE
Indent rule - flatTernaryExpressions set to true falsely reporting errors

### ISSUE THREAD

**Issue by ghostrib** [eslint/eslint#13384]
**Tell us about your environment**

* **ESLint Version:** v7.1.0
* **Node Version:** v10.15.3
* **npm Version:** v6.4.1

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```json
{
    "root": true,
    "env": {
        "browser": true,
        "commonjs": true,
        "es6": true,
        "node": true
    },
    "extends": [
        "standard"
    ],
    "globals": {
        "Atomics": "readonly",
        "SharedArrayBuffer": "readonly"
    },
    "parserOptions": {
        "ecmaVersion": 11
    },
    "rules": {
        "array-bracket-spacing":["error", "always"],
        "semi":["error", "always"],
        "brace-style": ["error", "stroustrup"],
        "indent": ["error", 2, {
            "SwitchCase": 1,
            "flatTernaryExpressions": true
          }
        ]
    }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

**[Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgaW5kZW50OiBbXCJlcnJvclwiLCAyLCB7IFwiZmxhdFRlcm5hcnlFeHByZXNzaW9uc1wiOiB0cnVlIH1dKi9cblxuY29uc3QgdGVzdCA9IChhLCBiKSA9PiB7XG4gIGNvbnN0IHJlc3VsdCA9IGEubGVuZ3RoID4gYi5sZW5ndGhcbiAgICA/ICdBIGlzIGxvbmdlciB0aGFuIEInXG4gICAgOiBhLmxlbmd0aCA8IGIubGVuZ3RoXG4gICAgPyAnQiBpcyBsb25nZXIgdGhhbiBBJ1xuICAgIDogJ0EgYW5kIEIgaGF2ZSBlcXVhbCBsZW5ndGgnO1xuICByZXR1cm4gcmVzdWx0O1xufTtcblxudGVzdCgnaGVsbG8nLCAnZ29vZGJ5ZScpOyIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6MTEsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnsiaW5kZW50IjoyfSwiZW52Ijp7fX19)**

```js
const test = (a, b) => {
  const result = a.length > b.length
    ? 'A is longer than B'
    : a.length < b.length
    ? 'B is longer than A'
    : 'A and B have equal length';
  return result;
};

test('hello', 'goodbye');

```

```bash
./node_modules/.bin/eslint ./src/index.js 
```

**What did you expect to happen?**
Expected to not receive any errors in regards to indentation.

**What actually happened? Please include the actual, raw output from ESLint.**
2 errors reported regarding indentation.
```bash
  5:1  error  Expected indentation of 6 spaces but found 4  indent
  6:1  error  Expected indentation of 6 spaces but found 4  indent

✖ 2 problems (2 errors, 0 warnings)
  2 errors and 0 warnings potentially fixable with the `--fix` option.
```

**Are you willing to submit a pull request to fix this bug?**
I don't have a fix at the moment but if I find one then yes

**Comment by anikethsaha** [comment]
May be because the colons (`:`) are the leading instead of trailing ?

[I guess this  is the explanation 🤔 ](https://github.com/eslint/eslint/blob/master/lib/rules/indent.js#L1176-L1189)

[demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgaW5kZW50OiBbXCJlcnJvclwiLCAyLCB7IFwiZmxhdFRlcm5hcnlFeHByZXNzaW9uc1wiOiB0cnVlIH1dKi9cblxuY29uc3QgdGVzdCA9IChhLCBiKSA9PiB7XG4gIGNvbnN0IHJlc3VsdCA9IFxuICAgIGEubGVuZ3RoID4gYi5sZW5ndGggPyAnQSBpcyBsb25nZXIgdGhhbiBCJzogXG4gIFx0YS5sZW5ndGggPCBiLmxlbmd0aCA/ICdCIGlzIGxvbmdlciB0aGFuIEEnOiBcbiAgICAnQSBhbmQgQiBoYXZlIGVxdWFsIGxlbmd0aCc7XG4gIHJldHVybiByZXN1bHQ7XG59O1xuXG50ZXN0KCdoZWxsbycsICdnb29kYnllJyk7XG4iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjExLCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7ImluZGVudCI6Mn0sImVudiI6e319fQ==)


Not sure though.

**Comment by ghostrib** [comment]
I think you're right. The label "flatTernaryExpressions" is a little confusing then.
